gateway/ln-gateway/src/state_machine/pay.rs

```diff
@@ -531,21 +532,18 @@ impl GatewayPayInvoice {
         contract: OutgoingContractAccount,
     ) -> Result<(), OutgoingPaymentError> {
         let mut dbtx = context.gateway.gateway_db.begin_transaction().await;
-        if let Some(secret_hash) = dbtx
-            .get_value(&PreimageAuthentication { payment_hash })
-            .await
-        {
+        if let Some(secret_hash) = dbtx.load_preimage_authentication(payment_hash).await {
             if secret_hash != preimage_auth {
                 return Err(OutgoingPaymentError {
-                    error_type: OutgoingPaymentErrorType::InvoiceAlreadyPaid,
+                    error_type: OutgoingPaymentErrorType::InvalidInvoicePreimage,
```

Comment:
Note for reviewers: Renaming this from `InvoiceAlreadyPaid` to `InvalidInvoicePreimage` is unrelated to the rest of the changes here. I happened to notice that the existing error type didn't fit.